apps/bmd/src/AppRoot.tsx

```diff
@@ -141,898 +147,1091 @@ export const activationStorageKey = 'activation'
+  }, [card])
+
+  const writeCard = useCallback(
+    async (cardData: VoterCardData) => {
+      await card.writeShortValue(JSON.stringify(cardData))
+    },
+    [card]
+  )
+
+  // Handle Storing Election Locally
+  useEffect(() => {
+    const storeElection = (electionDefinition: ElectionDefinition) => {
+      storage.set(electionStorageKey, electionDefinition)
+    }
+    if (optionalElectionDefinition) {
+      storeElection(optionalElectionDefinition)
+    }
+  }, [optionalElectionDefinition, storage])
+
+  // Handle Ballot Activation
+  useEffect(() => {
+    const storeBallotActivation = (data: SerializableActivationData) => {
+      /* istanbul ignore else */
+      if (process.env.NODE_ENV !== 'production') {
+        storage.set(activationStorageKey, data)
+      }
+    }
 
-  public toggleLiveMode = (): void => {
-    this.setState(
-      (prevState) => ({
-        isLiveMode: !prevState.isLiveMode,
-        isPollsOpen: this.initialAppState.isPollsOpen,
-      }),
-      this.resetTally
-    )
-  }
+    if (precinctId && ballotStyleId) {
+      storeBallotActivation({
+        ballotCreatedAt: voterCardCreatedAt,
+        precinctId,
+        ballotStyleId,
+      })
+    }
+  }, [precinctId, ballotStyleId, voterCardCreatedAt, storage])
 
-  public togglePollsOpen = (): void => {
-    this.setState(
-      (prevState) => ({ isPollsOpen: !prevState.isPollsOpen }),
-      this.storeAppState
-    )
-  }
+  // Handle Vote Updated (and store votes locally in !production)
+  useEffect(() => {
+    const storeVotes = async (votes: VotesDict) => {
+      const storedVotes = storage.get(votesStorageKey) || blankBallotVotes
+      if (JSON.stringify(storedVotes) !== JSON.stringify(votes)) {
+        /* istanbul ignore else */
+        if (process.env.NODE_ENV !== 'production') {
+          storage.set(votesStorageKey, votes)
+        }
 
-  public resetTally = (): void => {
-    this.setState(
-      ({ electionDefinition }) => ({
-        ballotsPrintedCount: this.initialAppState.ballotsPrintedCount,
-        tally: getZeroTally(electionDefinition!.election),
-      }),
-      this.storeAppState
-    )
-  }
+        dispatchState({ type: 'updateLastVoteUpdateAt', date: Date.now() })
+      }
+    }
+    if (votes) {
+      storeVotes(votes)
+    }
+  }, [votes, storage])
+
+  const resetBallot = useCallback(
+    (path = '/') => {
+      storage.remove(activationStorageKey)
+      storage.remove(votesStorageKey)
+      dispatchState({ type: 'resetBallot' })
+      history.push(path)
+    },
+    [storage, history]
+  )
+
+  const unconfigure = useCallback(() => {
+    dispatchState({ type: 'unconfigure' })
+    storage.clear()
+    history.push('/')
+  }, [storage, history])
+
+  const updateVote = useCallback((contestId: string, vote: OptionalVote) => {
+    dispatchState({ type: 'updateVote', contestId, vote })
+  }, [])
+
+  const forceSaveVote = useCallback(() => {
+    dispatchState({ type: 'forceSaveVoteFlag', value: true })
```

Comment:
Would this make more sense as `setForceSaveVoteFlag`? Or better yet, `forceSaveVote` with no value since we're never calling this with `false`?

Reply:
Agreed. Standardized on `forceSaveVote` and removed the passed value.